PWGHF/TableProducer/treeCreatorDsToKKPi.cxx

```diff
@@ -255,8 +271,28 @@ struct HfTreeCreatorDsToKKPi {
   Partition<CandDsMcReco> reconstructedCandSig = nabs(aod::hf_cand_3prong::flagMcMatchRec) == static_cast<int8_t>(BIT(aod::hf_cand_3prong::DecayType::DsToKKPi)) && (aod::hf_cand_3prong::flagMcDecayChanRec == decayChannel || (fillDplusMc && aod::hf_cand_3prong::flagMcDecayChanRec == (decayChannel + offsetDplusDecayChannel))); // Do not store Dplus MC if fillDplusMc is false
   Partition<CandDsMcReco> reconstructedCandBkg = nabs(aod::hf_cand_3prong::flagMcMatchRec) != static_cast<int8_t>(BIT(aod::hf_cand_3prong::DecayType::DsToKKPi));
 
+  // CCDB service
+  Service<o2::ccdb::BasicCCDBManager> ccdb;
+  o2::base::Propagator::MatCorrType noMatCorr = o2::base::Propagator::MatCorrType::USEMatCorrNONE;
+  int runNumber;
+  double bz{0.};
+
   void init(InitContext const&)
   {
+    // Configure CCDB access
+    ccdb->setURL(ccdbUrl);
+    ccdb->setCaching(true);
+    ccdb->setLocalObjectValidityChecking();
+    runNumber = 0;
+  }
+
+  template <typename T1, typename T2>
+  float getProngDcaXY(const T1& prong, const T2& collision)
+  {
+    auto trackPar = getTrackPar(prong);
+    o2::gpu::gpustd::array<float, 2> dcaInfo{-999., -999.};
+    o2::base::Propagator::Instance()->propagateToDCABxByBz({collision.posX(), collision.posY(), collision.posZ()}, trackPar, 2.f, noMatCorr, &dcaInfo);
+    return dcaInfo[0];
```

Comment:
This is in principle necessary only for tracks reassociated by the `track-to-collision-associator`, otherwise for tracks automatically grouped to collisions this should have been already provided by the `track-propagation` workflow. Are you really using reassociated tracks and calling this function only for those tracks? Looking at your code it does not seems to me, but correct me if I'm wrong.